plugin/src/main/java/dev/regadas/trino/pubsub/listener/CompressingMessageEncoder.java

```diff
@@ -0,0 +1,40 @@
+package dev.regadas.trino.pubsub.listener;
+
+import com.google.protobuf.Message;
+import dev.regadas.trino.pubsub.listener.Encoder.MessageEncoder;
+import java.io.ByteArrayOutputStream;
+import java.util.Objects;
+import java.util.zip.Deflater;
+
+public class CompressingMessageEncoder implements MessageEncoder {
+
+    private static final int BUFFER_SIZE = 32 * 1024;
+    private static final int COMPRESSION_LEVEL = Deflater.BEST_COMPRESSION;
```

Comment:
So if I'm not mistaken this will use zlib. I think we should bring zstd into the mix as compression/decompression ratio and speed are far better.

Reply:
Correct, I didn't want to include external dependencies but I can change it